eip_operator/src/controller/eip.rs

```diff
@@ -80,13 +93,207 @@ impl k8s_controller::Context for Context {
                     .public_ip
                     .as_ref()
                     .ok_or(Error::MissingPublicIp)?;
-                (allocation_id.to_owned(), public_ip.to_owned())
+                status.allocation_id = Some(allocation_id.clone());
+                status.public_ip_address = Some(public_ip.clone());
             }
             _ => {
                 return Err(Error::MultipleEipsTaggedForPod);
             }
         };
-        crate::eip::set_status_created(&eip_api, name, &allocation_id, &public_ip).await?;
+
+        // get potential attachments
+        let mut matched_pods: Vec<Pod> = vec![];
+        let mut matched_nodes: Vec<Node> = vec![];
+
+        match eip.spec.selector {
+            EipSelector::Pod { ref pod_name } => {
```

Comment:
is `ref` actually necessary here?

Reply:
At one point my LSP told me to do it so I did it. I'm not sure if it's still needed, but nothing is complaining.